test/extended/builds/ldap_fixtures.go

```diff
@@ -0,0 +1,90 @@
+		baseImageStreamName      = "openldap"
+		imageStreamFixture       = exutil.FixturePath("fixtures", "ldap", "ldapserver-imagestream.json")
+		imageStreamTargetFixture = exutil.FixturePath("fixtures", "ldap", "ldapserver-imagestream-testenv.json")
+		buildConfigFixture       = exutil.FixturePath("fixtures", "ldap", "ldapserver-buildconfig.json")
+		deploymentConfigFixture  = exutil.FixturePath("fixtures", "ldap", "ldapserver-deploymentconfig.json")
+		serviceConfigFixture     = exutil.FixturePath("fixtures", "ldap", "ldapserver-service.json")
+		oc                       = exutil.NewCLI("openldap", exutil.KubeConfigPath())
+	)
+
+	Describe("Building and deploying an OpenLDAP server", func() {
+		It(fmt.Sprintf("should create a image from %q template and run it in a pod", buildConfigFixture), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			By(fmt.Sprintf("calling oc create -f %q", imageStreamFixture))
+			err := oc.Run("create").Args("-f", imageStreamFixture).Execute()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("expecting the imagestream to fetch and tag the latest image")
+			err = exutil.WaitForImageStream(oc.REST().ImageStreams(oc.Namespace()), baseImageStreamName,
+				// The imagestream has a ':latest' tag filed
+				func(i *imageapi.ImageStream) bool {
+					_, ok := i.Status.Tags["latest"]
+					return ok
+				},
+				// The imagestream could not find the respository
+				func(i *imageapi.ImageStream) bool {
+					return strings.Contains(i.Annotations[imageapi.DockerImageRepositoryCheckAnnotation], "was not found")
+				},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("calling oc create -f %q", imageStreamTargetFixture))
+			err = oc.Run("create").Args("-f", imageStreamTargetFixture).Execute()
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("calling oc create -f %q", buildConfigFixture))
+			err = oc.Run("create").Args("-f", buildConfigFixture).Execute()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("starting the test environment image build")
+			buildName, err := oc.Run("start-build").Args("openldap").Output()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("expecting the build to be in Complete phase")
+			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName,
+				// The build passed
+				func(b *buildapi.Build) bool {
+					return b.Name == buildName && b.Status.Phase == buildapi.BuildPhaseComplete
+				},
+				// The build failed
+				func(b *buildapi.Build) bool {
+					if b.Name != buildName {
+						return false
+					}
+					return b.Status.Phase == buildapi.BuildPhaseFailed || b.Status.Phase == buildapi.BuildPhaseError
+				},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("calling oc create -f %q", deploymentConfigFixture))
+			err = oc.Run("create").Args("-f", deploymentConfigFixture).Execute()
+			Expect(err).NotTo(HaveOccurred())
+
+			By("starting the test environment deployment")
+			err = oc.Run("deploy").Args("openldap-server").Execute()
```

Comment:
This only displays the deployment.  It doesn't actually wait for it to complete.  Wait for it to finish.